[editor] Limit more fields (website, menu, social medias, etc...) for some POI types

Arthur-GYT opened the pull request:
All is in the title ^^.

Related to https://github.com/organicmaps/organicmaps/pull/8052#issuecomment-2118874974

Jean-BaptisteC commented:
The right solution is to update editor.config

Arthur-GYT commented:
@Jean-BaptisteC It's strange. I test it with restaurant and it appear for haircut. Do I miss something ?

biodranik commented on data/editor.config:
```diff
@@ -354,6 +354,7 @@
       <type id="amenity-restaurant" group="food">
         <include group="poi_internet" />
         <include field="cuisine" />
+        <include field="website_menu" />
```
Please review all related POIs at least in group=food, and also add missing cuisine tags if necessary.

Why was it displayed for Parking?

Arthur-GYT replied:
I forgot to add the field in details pane

Arthur-GYT replied:
I think it's done. Also, I don't see missing cuisine tags.

Arthur-GYT commented:
@biodranik  I've changed the scope of the PR. I've also fixed the fact that, for example, parking lots were displaying fields for social medias in the editor (which doesn't make sense).

biodranik commented on android/app/src/main/java/app/organicmaps/editor/EditorFragment.java:
```diff
@@ -469,6 +469,16 @@ private void initViews(View view)
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_WEBSITE_MENU, websiteMenuBlock);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_EMAIL, emailBlock);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_CONTACT_FACEBOOK, facebookContactBlock);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_CONTACT_INSTAGRAM, instagramContactBlock);
```
How did it work before?

Arthur-GYT replied:
Previously, it ignored the configuration described in editor.config. You need to add the field as shown here to make it take this configuration into account.

![Screenshot_20240519-084611.png](https://github.com/organicmaps/organicmaps/assets/116185937/8e39b5e6-29da-4afb-9e4f-d41714980354)

Arthur-GYT replied:
@biodranik Do I need to change anything?

biodranik commented on android/app/src/main/java/app/organicmaps/editor/EditorFragment.java:
```diff
@@ -469,6 +469,16 @@ private void initViews(View view)
     mDetailsBlocks.put(Metadata.MetadataType.FMD_INTERNET, blockWifi);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_WEBSITE, websiteBlock);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_WEBSITE_MENU, websiteMenuBlock);
+    mDetailsBlocks.put(Metadata.MetadataType.FMD_WEBSITE_MENU, websiteMenuBlock);
```
Copy-paste from above line?

Arthur-GYT replied:
Fixed ^^.

biodranik commented on data/editor.config:
```diff
@@ -1051,6 +1056,7 @@
       </type>
       <type id="amenity-biergarten" group="food">
         <include group="poi_internet" />
+        <include field="website_menu" />
```
Does it make sense to allow editing cuisine for a biergarten? @organicmaps/translations-de WDYT?

Arthur-GYT replied:
According to the wiki, it make sense ^^ 

https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbiergarten

biodranik reviewed (approved):
LGTM, Did you test it? Note that debug build of OM uploads data into the dev server, and you need a separate account there:

https://master.apis.dev.openstreetmap.org/

If you need to edit some data from the main OSM server, you can upload it to the dev server using this tool: https://github.com/Zverik/osm_to_sandbox

@Jean-BaptisteC PTAL if you have time.

Jean-BaptisteC reviewed (approved):
✅ Pixel 6 - Android 14
https://master.apis.dev.openstreetmap.org/node/4349085700
